### PR TITLE
Refactor sanity checks to use pytest

### PR DESCRIPTION
**Pull request by mreso**
## Description

This PR refactors parts of our sanity checks to use pytest infra which enables devs to test model registration + snapshotting using
```
pytest test/pytest/sanity
```

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

## Feature/Issue validation/testing

Please describe the Unit or Integration tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [X] pytest test/pytest/sanity
```
========================================================================================================================================================================== test session starts ===========================================================================================================================================================================
platform linux -- Python 3.9.16, pytest-7.2.2, pluggy-1.0.0
rootdir: /home/ubuntu/serve
plugins: mock-3.10.0, cov-4.0.0
collected 22 items

test/pytest/sanity/test_config_snapshot.py .                                                                                                                                                                                                                                                                                                                       [  4%]
test/pytest/sanity/test_model_registering.py .....................                                                                                                                                                                                                                                                                                                 [100%]/home/ubuntu/miniconda3/envs/microbatching/lib/python3.9/site-packages/coverage/control.py:858: CoverageWarning: No data was collected. (no-data-collected)
  self._warn("No data was collected.", slug="no-data-collected")


---------- coverage: platform linux, python 3.9.16-final-0 -----------
Coverage XML written to file test/pytest/sanity/coverage.xml


===================================================================================================================================================================== 22 passed in 307.21s (0:05:07) =====================================================================================================================================================================
```

- [X] python torchserve_sanity.py
See PR checks


## Checklist:

- [X] Did you have fun?
- [X] Have you added tests that prove your fix is effective or that this feature works?
- [X] Has code been commented, particularly in hard-to-understand areas?

### CODE REVIEW

File: ts_scripts/sanity_utils.py [hunk 1]
```diff
@@ -163,51 +162,18 @@ def run_rest_test(model, register_model=True, unregister_model=True):
 
 
 def test_sanity():
-    generate_grpc_client_stubs()
```
Comment (lxning) [inline ts_scripts/sanity_utils.py:166]: I guess sanity test will fail without gRPC client.

**Review by lxning — COMMENTED** [review]
Currently regression test run [test/pytest](https://github.com/pytorch/serve/blob/master/ts_scripts/regression_utils.py#L53). 

With these changes, tt seems that regression test will also run test_sanity. Do we need this duplicate work?

File: ts_scripts/sanity_utils.py (hunk 1 above)
Comment (mreso) [reply ts_scripts/sanity_utils.py:207]: Thats call is now a fixture https://github.com/pytorch/serve/blob/15a9fd9abd703a0291a87a80b46fe06c538373aa/test/pytest/sanity/test_model_registering.py#L15

**Comment by mreso** [comment]
> Currently regression test run [test/pytest](https://github.com/pytorch/serve/blob/master/ts_scripts/regression_utils.py#L53).
> 
> With these changes, tt seems that regression test will also run test_sanity. Do we need this duplicate work?

@lxning good point! Have changes the PR so that sanity tests are ignored when regression tests run

**Review by lxning — APPROVED** [review]

**Pull request merged**